src/configuration.rs

```diff
@@ -561,4 +966,210 @@ mod test {
+                operator: WhenConditionOperator::Equal,
+                value: "\"123\"".to_string(),
+                path: Default::default(),
+                compiled: Default::default(),
+            };
+            p.compile().expect("Should compile fine!");
+            assert_eq!(
+                p.eval("123".as_bytes().to_vec()),
+                Ok(true),
+                "Expression: {:?}",
+                p.compiled.get()
+            )
+        }
+
+        #[test]
+        fn test_int() {
+            let p = PatternExpression {
+                selector: "destination.port".to_string(),
+                operator: WhenConditionOperator::Equal,
+                value: "8080".to_string(),
+                path: Default::default(),
+                compiled: Default::default(),
+            };
+            p.compile().expect("Should compile fine!");
+            assert_eq!(
+                p.eval(8080_i64.to_le_bytes().to_vec()),
+                Ok(true),
+                "Expression: {:?}",
+                p.compiled.get()
+            )
+        }
+
+        #[test]
+        fn test_int_inferred() {
+            let p = PatternExpression {
+                selector: "foobar".to_string(),
+                operator: WhenConditionOperator::Equal,
+                value: "8080".to_string(),
+                path: Default::default(),
+                compiled: Default::default(),
+            };
+            p.compile().expect("Should compile fine!");
+            assert_eq!(
+                p.eval(8080_i64.to_le_bytes().to_vec()),
+                Ok(true),
+                "Expression: {:?}",
+                p.compiled.get()
+            )
+        }
+
+        #[test]
+        fn test_float_inferred() {
+            let p = PatternExpression {
+                selector: "foobar".to_string(),
+                operator: WhenConditionOperator::Equal,
+                value: "1.0".to_string(),
+                path: Default::default(),
+                compiled: Default::default(),
+            };
+            p.compile().expect("Should compile fine!");
+            assert_eq!(
+                p.eval(1_f64.to_le_bytes().to_vec()),
+                Ok(true),
+                "Expression: {:?}",
+                p.compiled.get()
+            )
+        }
+
+        #[test]
+        fn test_bool() {
+            let p = PatternExpression {
+                selector: "connection.mtls".to_string(),
+                operator: WhenConditionOperator::Equal,
+                value: "true".to_string(),
+                path: Default::default(),
+                compiled: Default::default(),
+            };
+            p.compile().expect("Should compile fine!");
+            assert_eq!(
+                p.eval((true as u8).to_le_bytes().to_vec()),
+                Ok(true),
+                "Expression: {:?}",
+                p.compiled.get()
+            )
+        }
+
+        #[test]
+        #[ignore]
```

Comment:
I think this might require another [fix](https://github.com/clarkmcc/cel-rust/issues/67) but not a huge deal, as I don't think that's really useful right now anyways…